shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -765,9 +765,11 @@ def post_expansion(self, constraints: list["Constraint"]) -> None:
         self.rhs.index = self.rhs_index
         self.acc.index = self.acc_index
 
-        if self.lhs_type.address_space == SHARED_ADDRESS_SPACE:
+        # TODO: this is really wrong place for it, it relies on specific kernel structure,
+        # generated, if mma input is not come from load, things will breaks.
```

Comment:
Yes good point. With the separation into type and memory_type, not all ops will have a memory_type and so this won't work. Can you move this into the index setter for Read and Write Ops which I think will be a better place for this? So the idea is that we set the lhs, rhs and acc indices here. But within the Read and Write index setters, we can check if the memory type has a shared address space and if so we remove the global indexing.

Reply:
I tried to make index setter smart, but it didn't wok., `remove_global_indexing` need access to constraints, which are not available in setter.

Reply:
Thanks for trying, I can take a look at this.